alerting/notifier/channels/kafka.go

```diff
@@ -130,11 +207,108 @@ func (kn *KafkaNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool,
 		},
 	}
 
-	if err = kn.ns.SendWebhook(ctx, cmd); err != nil {
-		kn.log.Error("Failed to send notification to Kafka", "error", err, "body", body)
+	retryOnFailure := false
+	if kn.settings.BasicAuth {
+		cmd.User = kn.settings.BasicAuthUser
+		cmd.Password = kn.settings.AuthPass
+
+		// Only retry if the password is read from a file becase the password could have changed.
+		retryOnFailure = kn.settings.PasswordSource == PasswordSourceFromFile
+	}
+
+	return kn.sendWebhookWithRetry(ctx, cmd, retryOnFailure)
+}
+
+// Use the v3 API to send the alert notification.
+func (kn *KafkaNotifier) notifyWithAPIV3(ctx context.Context, as ...*types.Alert) (bool, error) {
+	var tmplErr error
+	tmpl, _ := TmplText(ctx, kn.tmpl, as, kn.log, &tmplErr)
+
+	// For v3 the Produce URL is like this,
+	// <Endpoint>/kafka/v3/clusters/<Cluster ID>/topics/<Topic Name>/records
+	topicURL := strings.TrimRight(kn.settings.Endpoint, "/") + "/kafka/v3/clusters/" + tmpl(kn.settings.KafkaClusterID) + "/topics/" + tmpl(kn.settings.Topic) + "/records"
+
```

Comment:
can you add check here
```
if tmplErr != nil {
		kn.log.Warn("failed to template Kafka URL", "error", tmplErr.Error())
	}
```

Reply:
Sorry, there was a check here, I added additional code in the middle and it got moved down to line 235. I'll fix this!

Reply:
Done!